src/framework/simple/mod.rs

```diff
@@ -149,35 +156,52 @@ impl SimpleFramework {
         }
     }
 
+    async fn run_prefix_only_cmd(&self, ctx: &Context, msg: &Message) {
+        if let Some(prefix_only) = &self.prefix_only_cmd {
+            prefix_only(ctx, msg).await;
+        }
+    }
+
+    async fn run_normal_message(&self, ctx: &Context, msg: &Message) {
+        if let Some(normal_message) = &self.normal_message_fn {
+            normal_message(ctx, msg).await;
+        }
+    }
+
+
+    fn parse_cmd_name_and_args<'a>(&self, text: &'a str) -> (Cow<'a, str>, Args) {
+        let text = match &self.delimiters[0] {
+            Delimiter::Multiple(delim) => text.splitn(2, delim).collect::<Vec<&str>>(),
+            Delimiter::Single(delim) => text.splitn(2, *delim).collect::<Vec<&str>>(),
+        };
+
+        let cmd = if self.case_insensitive {
+            Cow::Owned(text[0].to_lowercase())
+        } else {
+            Cow::Borrowed(text[0])
+        };
+        let args = Args::new(text[1], &self.delimiters);
+        (cmd, args)
+    }
+
 }
 
 #[async_trait]
 impl Framework for SimpleFramework {
     async fn dispatch(&self, ctx: Context, msg: Message) {
-        if msg.content.starts_with(&self.prefix) {            
-            let cmd_and_args = msg.content.splitn(2, &self.prefix).collect::<Vec<&str>>();
-            if cmd_and_args[1].is_empty() {
 
-                if let Some(prefix_only) = &self.prefix_only_cmd {
-                    prefix_only(&ctx, &msg).await;
-                    return;
-                }
+        let text = msg.content.trim_start();
 
+        if !text.starts_with(&self.prefix) {
+            self.run_normal_message(&ctx, &msg).await;
+        } else {
```

Comment:
I prefer flat code when inside of a function. What I mean by "flat" code is that there's minimal indentation, so it doesn't vigorously stretch to the right; otherwise we get ["hadouken" code](https://external-preview.redd.it/6TTm4tS5pMCiq2KlgviubpGKt1QzvetOi2Z6Z3lB1Nc.jpg?auto=webp&s=c1df961d84c8d76048a93c1d261e425d3737d779) if left unchecked. Early returns can help avoid the latter. They also split the erroneous cases from the successful case, which will happen at the end of the function.
```suggestion
        if !text.starts_with(&self.prefix) {
            self.run_normal_message(&ctx, &msg).await;
            return;
        }
```